### PR TITLE
recursor: Add options to tune udp source ports

### PR DESCRIPTION
**Pull request by chbruyand**
Add min-udp-source-port, max-udp-source-port and avoid-udp-source-port variables to tune the range of ports we use

This should address #6321 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -3516,6 +3548,10 @@ int main(int argc, char **argv)
     ::arg().set("xpf-allow-from","XPF information is only processed from these subnets")="";
     ::arg().set("xpf-rr-code","XPF option code to use")="0";
 
+    ::arg().set("min-udp-source-port", "Minimum UDP port to bind on")="1025";
+    ::arg().set("max-udp-source-port", "Maximum UDP port to bind on")="65535";
+    ::arg().set("avoid-udp-source-port", "List of comma separated UDP port number to avoid")="11211";
```
Comment (pieterlexis) [inline pdns/pdns_recursor.cc]: Small nit:

Perhaps name the settings `udp-source-port-$SETTING` so they are sorted nicely in the generated config file.

**Comment by rgacogne** [comment]
Probably needs some documentation :)

File: pdns/pdns_recursor.cc (hunk 1 above)
Comment (zeha) [reply pdns/pdns_recursor.cc]: this is kinda why i thought having a single udp-source-port setting would be a lot nicer and maybe easier to understand

**Comment by krionbsd** [comment]
Fails on fbsd with:

In file included from pdns_recursor.cc:63:
In file included from /usr/local/include/boost/container/flat_set.hpp:29:
In file included from /usr/local/include/boost/container/detail/flat_tree.hpp:30:
In file included from /usr/local/include/boost/container/vector.hpp:58:
In file included from /usr/local/include/boost/move/algo/adaptive_merge.hpp:16:
/usr/local/include/boost/move/algo/detail/adaptive_sort_merge.hpp:813:14: error: illegal initializer
      (only variables can be initialized)
   size_type L = size_type(last - first);
             ^
./recursor_cache.hh:45:11: note: expanded from macro 'L'
#define L theL()
          ^

**Comment by ahupowerdns** [comment]
the solution for @krionbsd's problem is to move the boost include all the way up so it happens before logger.hh gets included. It is a historical mistake that L...

**Comment by chbruyand** [comment]
I moved the include on top and renamed the settings.

It should build. @krionbsd, is that easy for you to test ?

**Comment by krionbsd** [comment]
I confirm, now it builds fine on fbsd head with clang 6.0.0, thank you, Charles-Henri!

**Pull request merged**